src/pool/worker.rs

```diff
@@ -19,21 +20,34 @@ where
     T: TaskCell + Send,
     R: Runner<TaskCell = T>,
 {
+    fn slow_pop(&mut self) -> Option<Pop<T>> {
+        // Wait some time before going to sleep, which is more expensive.
+        let mut spin = SpinWait::new();
+        loop {
+            if let Some(t) = self.local.pop() {
+                return Some(t);
+            }
+            if !spin.spin() {
+                break;
+            }
+        }
+        self.runner.pause(&mut self.local);
+        let t = self.local.pop_or_sleep();
+        if t.is_some() {
+            self.runner.resume(&mut self.local);
+        }
+        t
+    }
+
     pub fn run(mut self) {
         self.runner.start(&mut self.local);
         while !self.local.core().is_shutdown() {
             let task = match self.local.pop() {
```

Comment:
Can we just use the implementation of `slow_pop` here and skip this `pop`?

Reply:
Just to avoid additional function jump.

Reply:
Even now the compiler seems to inline `slow_pop`. (`slow_pop` does not apppear in symbol list)